homeassistant/components/vicare/climate.py

```diff
@@ -298,16 +298,27 @@ def set_preset_mode(self, preset_mode: str) -> None:
                 f"Cannot set invalid vicare program: {preset_mode}/{vicare_program}"
             )
 
-        _LOGGER.debug("Setting preset to %s / %s", preset_mode, vicare_program)
+        _LOGGER.debug("Current preset %s", self._current_program)
         if self._current_program != VICARE_PROGRAM_NORMAL:
             # We can't deactivate "normal"
+            _LOGGER.debug("deactivating %s", self._current_program)
             try:
                 self._circuit.deactivateProgram(self._current_program)
-            except PyViCareCommandError:
-                _LOGGER.debug("Unable to deactivate program %s", self._current_program)
+            except PyViCareCommandError as err:
+                raise ValueError(
```

Comment:
We now support translatable errors. Maybe we could add them here. What do you think?
See https://developers.home-assistant.io/docs/internationalization/core/#exceptions